packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -11,6 +13,117 @@ import 'binding.dart';
+  Future<void> handlePlatformMessage(String channel, ByteData? data, PlatformMessageResponseCallback? callback) {
+    return proxy.handlePlatformMessage(channel, data, callback);
+  }
+
+  Future<ByteData?>? sendWithPostfix(String channel, String postfix, ByteData? message) async {
+    final TimelineTask task = TimelineTask();
+    _debugRecordUpStream(channel, '$name$postfix', codec, message);
+    task.start('Platform Channel send $name$postfix');
+    final ByteData? result = await proxy.send(channel, message);
+    task.finish();
+    _debugRecordDownStream(channel, '$name$postfix', codec, result);
+    return result;
+  }
+
+  @override
+  Future<ByteData?>? send(String channel, ByteData? message) =>
+    sendWithPostfix(channel, '', message);
+
+  @override
+  void setMessageHandler(String channel, MessageHandler? handler) {
+    proxy.setMessageHandler(channel, handler);
+  }
+}
+
+class _PlatformChannelStats {
+  _PlatformChannelStats(this.channel, this.codec, this.type);
+
+  final String channel;
+  final String codec;
+  final String type;
+
+  int _upCount = 0;
+  int _upBytes = 0;
+  int get upBytes => _upBytes;
+  void addUpStream(int bytes) {
+    _upCount += 1;
+    _upBytes += bytes;
+  }
+
+  int _downCount = 0;
+  int _downBytes = 0;
+  int get downBytes => _downBytes;
+  void addDownStream(int bytes) {
+    _downCount += 1;
+    _downBytes += bytes;
+  }
+
+  double get averageUpPayload => _upBytes / _upCount;
+  double get averageDownPayload => _downBytes / _downCount;
+}
+
+final Map<String, _PlatformChannelStats> _debugProfilePlatformChannelsStats = <String, _PlatformChannelStats>{};
+
+Future<void> _debugLaunchProfilePlatformChannels() async {
+  if (!_debugProfilePlatformChannelsIsRunning) {
+    _debugProfilePlatformChannelsIsRunning = true;
+    await Future<dynamic>.delayed(_debugProfilePlatformChannelsRate);
+    _debugProfilePlatformChannelsIsRunning = false;
+    final StringBuffer log = StringBuffer();
+    log.writeln('Platform Channel Stats:');
+    final List<_PlatformChannelStats> allStats =
+        _debugProfilePlatformChannelsStats.values.toList();
+    // Sort highest combined bandwidth first.
+    allStats.sort((_PlatformChannelStats x, _PlatformChannelStats y) =>
+        (y.upBytes + y.downBytes) - (x.upBytes + x.downBytes));
+    for (final _PlatformChannelStats stats in allStats) {
+      log.writeln(
+          '  (name:${stats.channel} type:${stats.type} codec:${stats.codec} up:${stats.upBytes} up_avg:${stats.averageUpPayload} down:${stats.downBytes} down_avg:${stats.averageDownPayload})');
+    }
+    // ignore: avoid_print
+    print(log.toString());
+    _debugProfilePlatformChannelsStats.clear();
+  }
+}
+
+void _debugRecordUpStream(
+    Object channel, String name, Object codec, ByteData? bytes) {
+  final _PlatformChannelStats stats =
+      _debugProfilePlatformChannelsStats[name] ??=
+          _PlatformChannelStats(
+              name, codec.toString(), channel.runtimeType.toString());
```

Comment:
In the example in the PR description the codec looks something like:

```
Instance of 'StandardMethodCodec'
```

The "Instance of" is probably noise we don't need, can we implement toString on the codec to avoid that?

Reply:
Done